fish-rust/src/common.rs

```diff
@@ -89,5 +91,23 @@ pub fn escape_string(s: &wstr, style: EscapeStringStyle) -> WString {
         EscapeStringStyle::Regex => ffi::escape_string_style_t::STRING_STYLE_REGEX,
     };
 
-    ffi::escape_string(s.as_ptr(), flags_int.into(), style).from_ffi()
+    ffi::escape_string(c_str!(s), flags_int.into(), style).from_ffi()
```

Comment:
FWIW with `c_str!()` we have to be extra careful because it creates a reference to a temporary.
I believe in this case the temporary lives long enough.